src/svg-renderer.js

```diff
@@ -113,6 +123,32 @@ class SvgRenderer {
         };
     }
 
+    /**
+     * Load an SVG string, normalize it, and prepare it for (synchronous) rendering.
+     * @param {!string} svgString String of SVG data to draw in quirks-mode.
+     * @param {?boolean} fromVersion2 True if we should perform conversion from version 2 to version 3 svg.
+     * @param {Function} [onFinish] - An optional callback to call when the SVG is loaded and can be rendered.
+     */
+    loadSVG (svgString, fromVersion2, onFinish) {
+        this.loadString(svgString, fromVersion2);
+        this._createSVGImage(onFinish);
+    }
+
+    /**
+     * Creates an <img> element for the currently loaded SVG string, then calls the callback once it's loaded.
+     * @param {Function} [onFinish] - An optional callback to call when the <img> has loaded.
+     */
+    _createSVGImage (onFinish) {
+        const img = new Image();
+        this._cachedImage = null;
+        img.onload = () => {
+            this._cachedImage = img;
```

Comment:
It looks like this could get "confused" if two calls to `_createSVGImage` happen in quick succession -- maybe if the user is enthusiastically editing an SVG in the paint editor, for example. Do you think it's OK to have more than one of these image loads in flight?

In discussing this with @fsih we came across the idea of reusing the same `Image` element and (re)setting its `src` each time `_createSVGImage` is called. For example, instead of `const img = new Image();` and later `img.src = ...`, using `this._cachedImage.src = ...`. That might cancel previous loads, resulting in `onFinish` not being called. If we go with that solution, we'd need to make sure that it's OK for `onFinish` to be skipped for canceled loads. Adding a done/canceled parameter to `onFinish` is another idea. Do you think any of these are necessary?

Reply:
Don't we want `onFinish` to get skipped for canceled loads, so that the renderer doesn't get "confused"?

Reply:
Yes, we don't want to call the previous onFinish callback if this function gets called again with a new SVG string. One way to cancel the onload function would be to reuse the 'img' element. Then setting img.src = "" should prevent any previous onload function from being called.